File: backend/routes/user_manual.py
"""
User Manual / Help Documentation Module
Provides comprehensive user guides for all features
"""
from fastapi import APIRouter, HTTPException
from typing import Optional, List, Dict, Any

router = APIRouter(prefix="/help", tags=["User Manual"])


# =============================================================================
# COMPREHENSIVE USER MANUAL DATA
# =============================================================================

USER_MANUAL = {
    "overview": {
        "title": "Welcome to CreatorStudio AI",
        "description": "Your all-in-one AI-powered content creation platform for viral reels, stories, and more.",
        "quickStart": [
            "1. Sign up or log in to your account",
            "2. Check your credit balance in the top navigation",
            "3. Choose a feature from the dashboard",
            "4. Follow the on-screen instructions",
            "5. Download or share your creations"
        ]
    },
    
    "features": {
        "genstudio": {
            "title": "GenStudio AI",
            "description": "Create stunning AI-generated images and videos from text prompts.",
            "icon": "sparkles",
            "creditCost": "10-30 credits per generation",
            "subFeatures": {
                "text_to_image": {
                    "title": "Text to Image",
                    "description": "Generate high-quality images from text descriptions.",
                    "howToUse": [
                        "1. Navigate to GenStudio → Text to Image",
                        "2. Enter a detailed description of the image you want",
                        "3. (Optional) Add negative prompts to exclude unwanted elements",
                        "4. Select aspect ratio and style",
                        "5. Click 'Generate' and wait for your image",
                        "6. Download or regenerate as needed"
                    ],
                    "tips": [
                        "Be specific about colors, lighting, and composition",
                        "Use artistic references like 'watercolor style' or 'cinematic lighting'",
                        "Negative prompts help avoid unwanted elements"
                    ],
                    "creditCost": "10 credits"
                },
                "text_to_video": {
                    "title": "Text to Video",
                    "description": "Create AI-generated videos from text prompts using Sora 2.",
                    "howToUse": [
                        "1. Navigate to GenStudio → Text to Video",
                        "2. Describe the video scene you want",
                        "3. Select duration (4, 8, or 12 seconds)",
                        "4. Choose aspect ratio",
                        "5. Click 'Generate' - videos take 2-5 minutes",
                        "6. Download when complete"
                    ],
                    "tips": [
                        "Describe movement and action clearly",
                        "Shorter videos have higher quality",
                        "Use cinematic descriptions for better results"
                    ],
                    "creditCost": "25+ credits (varies by duration)"
                },
                "image_to_video": {
                    "title": "Image to Video",
                    "description": "Animate still images into video clips.",
                    "howToUse": [
                        "1. Navigate to GenStudio → Image to Video",
                        "2. Upload an image (JPEG, PNG)",
                        "3. Describe the motion you want",
                        "4. Select duration and settings",
                        "5. Click 'Generate' and wait",
                        "6. Download the animated video"
                    ],
                    "creditCost": "20+ credits"
                }
            }
        },
        
        "story_generator": {
            "title": "Kids Story Generator",
            "description": "Create personalized children's stories with illustrations.",
            "icon": "book-open",
            "creditCost": "10 credits per story",
            "howToUse": [
                "1. Navigate to Create Kids Story Pack",
                "2. Enter character names and ages",
                "3. Select genre and theme",
                "4. Choose age group (2-4, 4-7, 7-10, 10+)",
                "5. Add any special elements",
                "6. Click 'Generate Story'",
                "7. View your story with illustrations"
            ],
            "tips": [
                "Use names of children you know for personalization",
                "Select age-appropriate themes",
                "Generated stories include moral lessons"
            ]
        },
        
        "reel_generator": {
            "title": "Reel Script Generator",
            "description": "Create viral reel scripts in seconds.",
            "icon": "video",
            "creditCost": "10 credits per reel",
            "howToUse": [
                "1. Navigate to Generate Reel Script",
                "2. Enter your topic or niche",
                "3. Select target platform (Instagram, TikTok, YouTube)",
                "4. Choose tone (Funny, Educational, Motivational)",
                "5. Click 'Generate Script'",
                "6. Copy or download your script"
            ],
            "tips": [
                "Be specific about your niche",
                "Include trending topics for better engagement",
                "Scripts include hooks and CTAs"
            ]
        },
        
        "story_series": {
            "title": "Story Series Mode",
            "description": "Turn single stories into multi-episode series with consistent characters.",
            "icon": "film",
            "creditCost": "8-18 credits (based on episode count)",
            "howToUse": [
                "1. Navigate to Story Series",
                "2. Select an existing story OR enter a new summary",
                "3. Add character names",
                "4. Select theme (Adventure, Mystery, etc.)",
                "5. Choose episode count (3, 5, or 7)",
                "6. Click 'Generate Series'",
                "7. Download outline or script pack"
            ],
            "features": [
                "Consistent character development",
                "Scene beats for each episode",
                "Cliffhanger endings",
                "Next episode hooks",
                "Character Bible add-on"
            ],
            "creditCost": {
                "3 episodes": "8 credits",
                "5 episodes": "12 credits",
                "7 episodes": "18 credits",
                "Character Bible": "+5 credits"
            }
        },
        
        "challenge_generator": {
            "title": "Challenge Generator",
            "description": "Create 7-day or 30-day content challenges with hooks and hashtags.",
            "icon": "calendar",
            "creditCost": "6-15 credits",
            "howToUse": [
                "1. Navigate to Challenge Generator",
                "2. Select duration (7 or 30 days)",
                "3. Choose your niche (Luxury, Fitness, etc.)",
                "4. Select platform (Instagram, YouTube, TikTok)",
                "5. Set your goal (Followers, Leads, Sales)",
                "6. Adjust time per day slider",
                "7. Click 'Generate Challenge'",
                "8. View daily content plan",
                "9. Download CSV calendar"
            ],
            "includes": [
                "Daily hooks and CTAs",
                "Platform-optimized hashtags",
                "Posting time recommendations",
                "Content complexity guidance"
            ]
        },
        
        "tone_switcher": {
            "title": "Tone Switcher",
            "description": "Transform text into different emotional tones without AI costs.",
            "icon": "wand",
            "creditCost": "1-5 credits",
            "howToUse": [
                "1. Navigate to Tone Switcher",
                "2. Paste your original text",
                "3. Choose target tone:",
                "   - Funny (adds humor and emojis)",
                "   - Aggressive (bold and direct)",
                "   - Calm (gentle and soothing)",
                "   - Luxury (elegant and premium)",
                "   - Motivational (inspiring and uplifting)",
                "4. Adjust intensity slider",
                "5. Select length preference",
                "6. Try 'Free Preview' first",
                "7. Generate variations (1, 5, or 10)",
                "8. Copy or download results"
            ],
            "tips": [
                "Higher intensity = more dramatic changes",
                "Free preview lets you test before committing credits",
                "Great for social media captions and scripts"
            ]
        },
        
        "coloring_book": {
            "title": "Kids Coloring Book Creator",
            "description": "Create personalized printable coloring books from your stories.",
            "icon": "palette",
            "creditCost": "5-8 credits",
            "howToUse": [
                "1. Navigate to Kids Coloring Book",
                "2. Select a story from your library",
                "3. Choose creation mode:",
                "   - DIY Mode: Empty frames with prompts",
                "   - Photo Mode: Upload images for outline conversion",
                "4. Configure export settings:",
                "   - Page count (8, 10, or 12)",
                "   - Paper size (A4 or US Letter)",
                "   - Activity pages (optional)",
                "   - Personalized cover (add child's name)",
                "5. Click 'Export PDF'",
                "6. PDF downloads automatically"
            ],
            "privacyNote": "Images are processed locally in your browser. We never upload or store them.",
            "includes": [
                "Cover page with title",
                "Scene pages with illustrations",
                "Activity pages (optional)",
                "Moral/lesson page",
                "Completion certificate"
            ]
        },
        
        "creator_tools": {
            "title": "Creator Tools",
            "description": "15+ AI-powered tools for content creators.",
            "icon": "tools",
            "creditCost": "Varies by tool",
            "tools": [
                {
                    "name": "30-Day Content Calendar",
                    "description": "Generate a full month of content ideas"
                },
                {
                    "name": "Carousel Generator",
                    "description": "Create swipeable carousel content"
                },
                {
                    "name": "Hashtag Bank",
                    "description": "Generate niche-specific hashtags"
                },
                {
                    "name": "Thumbnail Generator",
                    "description": "Create eye-catching thumbnails"
                },
                {
                    "name": "Trending Topics",
                    "description": "Find what's trending in your niche"
                }
            ]
        }
    },
    
    "account": {
        "credits": {
            "title": "Understanding Credits",
            "description": "Credits are the currency used for all AI generations.",
            "howItWorks": [
                "Each feature costs a certain number of credits",
                "Credits are deducted only after successful generation",
                "Failed generations don't cost credits",
                "Check your balance in the top navigation"
            ],
            "howToGetCredits": [
                "Subscribe to a plan (Weekly, Monthly, Quarterly)",
                "Purchase top-up credit packs",
                "Earn bonus credits through referrals"
            ]
        },
        "subscription": {
            "title": "Subscription Plans",
            "description": "Choose a plan that fits your needs.",
            "plans": [
                {"name": "Weekly", "price": "₹99 / $4.99", "credits": 30},
                {"name": "Monthly", "price": "₹299 / $9.99", "credits": 100},
                {"name": "Quarterly", "price": "₹699 / $24.99", "credits": 350}
            ],
            "howToManage": [
                "Go to Dashboard → View Plans",
                "View current subscription status",
                "Cancel or reactivate auto-renewal",
                "Upgrade to a higher plan anytime"
            ]
        },
        "payments": {
            "title": "Payments & Billing",
            "description": "Secure payments powered by Cashfree.",
            "supported": [
                "UPI (Google Pay, PhonePe, Paytm)",
                "Credit/Debit Cards",
                "Net Banking",
                "International Cards (USD → INR conversion)"
            ],
            "refunds": [
                "Request refund within 7 days of purchase",
                "Unused credits are refundable",
                "Refunds processed in 5-7 business days"
            ]
        }
    },
    
    "troubleshooting": {
        "common_issues": [
            {
                "issue": "Generation taking too long",
                "solution": "Video generation can take 2-5 minutes. Check job status in your dashboard."
            },
            {
                "issue": "Download not working",
                "solution": "Try right-clicking and 'Save As'. Check your browser's download settings."
            },
            {
                "issue": "Credits not deducted",
                "solution": "Credits are reserved during generation and captured on success."
            },
            {
                "issue": "Image not displaying",
                "solution": "Refresh the page. If issue persists, regenerate the content."
            },
            {
                "issue": "Payment failed",
                "solution": "Check card details or try a different payment method. Contact support if needed."
            }
        ],
        "contact": {
            "email": "support@creatorstudio.ai",
            "responseTime": "Within 24 hours"
        }
    }
}


# =============================================================================
# ENDPOINTS
# =============================================================================

@router.get("/manual")
async def get_full_manual():
    """Get complete user manual"""
    return USER_MANUAL


@router.get("/manual/{section}")
async def get_manual_section(section: str):
    """Get specific section of user manual"""
    if section not in USER_MANUAL:
        raise HTTPException(status_code=404, detail="Section not found")
    return {section: USER_MANUAL[section]}


@router.get("/feature/{feature_id}")
async def get_feature_guide(feature_id: str):
    """Get guide for specific feature"""
    features = USER_MANUAL.get("features", {})
    
    if feature_id in features:
        return features[feature_id]
    
    # Check subfeatures
    for feature in features.values():
        if "subFeatures" in feature and feature_id in feature["subFeatures"]:
            return feature["subFeatures"][feature_id]
    
    raise HTTPException(status_code=404, detail="Feature guide not found")


@router.get("/search")
async def search_manual(q: str):
    """Search user manual"""
    results = []
    query = q.lower()
    
    def search_dict(d, path=""):
        for key, value in d.items():
            current_path = f"{path}/{key}" if path else key
            if isinstance(value, dict):
                search_dict(value, current_path)
            elif isinstance(value, str) and query in value.lower():
                results.append({
                    "path": current_path,
                    "content": value[:200],
                    "match": key
                })
            elif isinstance(value, list):
                for i, item in enumerate(value):
                    if isinstance(item, str) and query in item.lower():
                        results.append({
                            "path": f"{current_path}[{i}]",
                            "content": item[:200],
                            "match": key
                        })
    
    search_dict(USER_MANUAL)
    return {"results": results[:20], "query": q}


@router.get("/quick-start")
async def get_quick_start():
    """Get quick start guide"""
    return {
        "title": "Quick Start Guide",
        "steps": [
            {
                "step": 1,
                "title": "Create Your Account",
                "description": "Sign up with email or Google",
                "link": "/signup"
            },
            {
                "step": 2,
                "title": "Get Credits",
                "description": "Subscribe or purchase credits",
                "link": "/app/billing"
            },
            {
                "step": 3,
                "title": "Choose a Feature",
                "description": "Pick from our suite of AI tools",
                "link": "/app"
            },
            {
                "step": 4,
                "title": "Create & Download",
                "description": "Generate content and download",
                "link": None
            }
        ],
        "popularFeatures": [
            {"name": "Kids Story Generator", "link": "/app/stories"},
            {"name": "Reel Script Generator", "link": "/app/reel"},
            {"name": "Text to Image", "link": "/app/gen-studio/text-to-image"},
            {"name": "Tone Switcher", "link": "/app/tone-switcher"}
        ]
    }

File: frontend/src/pages/UserManual.js
import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { Button } from '../components/ui/button';
import {
  BookOpen, Search, ArrowLeft, ChevronRight, ChevronDown,
  Sparkles, Video, BookMarked, Film, Calendar, Wand2,
  Palette, CreditCard, HelpCircle, Settings, Shield,
  Download, Play, Image, FileText, Users, Zap
} from 'lucide-react';
import api from '../utils/api';

const FEATURE_ICONS = {
  story_generator: BookMarked,
  reel_generator: Video,
  story_series: Film,
  challenge_generator: Calendar,
  tone_switcher: Wand2,
  coloring_book: Palette,
  creator_tools: Settings
};

export default function UserManual() {
  const [manual, setManual] = useState(null);
  const [loading, setLoading] = useState(true);
  const [searchQuery, setSearchQuery] = useState('');
  const [searchResults, setSearchResults] = useState([]);
  const [expandedSections, setExpandedSections] = useState({});
  const [activeFeature, setActiveFeature] = useState(null);

  useEffect(() => {
    fetchManual();
  }, []);

  const fetchManual = async () => {
    try {
      const response = await api.get('/api/help/manual');
      setManual(response.data);
    } catch (error) {
      console.error('Failed to load manual:', error);
    } finally {
      setLoading(false);
    }
  };

  const handleSearch = async (query) => {
    setSearchQuery(query);
    if (query.length < 2) {
      setSearchResults([]);
      return;
    }
    
    try {
      const response = await api.get(`/api/help/search?q=${encodeURIComponent(query)}`);
      setSearchResults(response.data.results || []);
    } catch (error) {
      console.error('Search error:', error);
    }
  };

  const toggleSection = (section) => {
    setExpandedSections(prev => ({
      ...prev,
      [section]: !prev[section]
    }));
  };

  if (loading) {
    return (
      <div className="min-h-screen bg-gradient-to-b from-slate-950 via-indigo-950 to-slate-950 flex items-center justify-center">
        <div className="animate-spin rounded-full h-8 w-8 border-t-2 border-b-2 border-purple-500"></div>
      </div>
    );
  }

  return (
    <div className="min-h-screen bg-gradient-to-b from-slate-950 via-indigo-950 to-slate-950">
      {/* Header */}
      <header className="border-b border-slate-800/50 bg-slate-900/50 backdrop-blur-sm sticky top-0 z-50">
        <div className="max-w-6xl mx-auto px-4 py-4">
          <div className="flex items-center justify-between">
            <div className="flex items-center gap-4">
              <Link to="/app" className="text-slate-400 hover:text-white">
                <ArrowLeft className="w-5 h-5" />
              </Link>
              <div className="flex items-center gap-2">
                <BookOpen className="w-8 h-8 text-purple-400" />
                <div>
                  <h1 className="text-xl font-bold text-white">User Manual</h1>
                  <p className="text-xs text-slate-400">Complete guide to Visionary Suite</p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>

      <main className="max-w-6xl mx-auto px-4 py-8">
        {/* Search */}
        <div className="mb-8">
          <div className="relative">
            <Search className="absolute left-4 top-1/2 transform -translate-y-1/2 w-5 h-5 text-slate-400" />
            <input
              type="text"
              value={searchQuery}
              onChange={(e) => handleSearch(e.target.value)}
              placeholder="Search the manual..."
              className="w-full bg-slate-800 border border-slate-700 rounded-xl pl-12 pr-4 py-3 text-white focus:border-purple-500 focus:outline-none"
            />
          </div>
          
          {searchResults.length > 0 && (
            <div className="mt-2 bg-slate-800 border border-slate-700 rounded-xl p-4">
              <h4 className="text-sm font-medium text-slate-400 mb-3">Search Results</h4>
              {searchResults.map((result, idx) => (
                <div key={idx} className="py-2 border-b border-slate-700 last:border-0">
                  <p className="text-sm text-white">{result.content}</p>
                  <p className="text-xs text-slate-500">{result.path}</p>
                </div>
              ))}
            </div>
          )}
        </div>

        {/* Quick Start */}
        <div className="mb-8 bg-gradient-to-r from-purple-500/10 to-pink-500/10 border border-purple-500/30 rounded-xl p-6">
          <h2 className="text-xl font-bold text-white mb-4 flex items-center gap-2">
            <Zap className="w-5 h-5 text-yellow-400" />
            Quick Start
          </h2>
          <div className="grid md:grid-cols-4 gap-4">
            {manual?.overview?.quickStart?.map((step, idx) => (
              <div key={idx} className="bg-slate-800/50 rounded-lg p-4">
                <p className="text-sm text-slate-300">{step}</p>
              </div>
            ))}
          </div>
        </div>

        {/* Features */}
        <div className="space-y-4">
          <h2 className="text-xl font-bold text-white mb-4">Feature Guides</h2>
          
          {manual?.features && Object.entries(manual.features).map(([key, feature]) => {
            const Icon = FEATURE_ICONS[key] || HelpCircle;
            const isExpanded = expandedSections[key];
            
            return (
              <div key={key} className="bg-slate-900/50 border border-slate-800 rounded-xl overflow-hidden">
                <button
                  onClick={() => toggleSection(key)}
                  className="w-full p-4 flex items-center justify-between hover:bg-slate-800/50 transition-colors"
                >
                  <div className="flex items-center gap-3">
                    <div className="w-10 h-10 rounded-lg bg-purple-500/20 flex items-center justify-center">
                      <Icon className="w-5 h-5 text-purple-400" />
                    </div>
                    <div className="text-left">
                      <h3 className="font-semibold text-white">{feature.title}</h3>
                      <p className="text-xs text-slate-400">{feature.description}</p>
                    </div>
                  </div>
                  {isExpanded ? (
                    <ChevronDown className="w-5 h-5 text-slate-400" />
                  ) : (
                    <ChevronRight className="w-5 h-5 text-slate-400" />
                  )}
                </button>
                
                {isExpanded && (
                  <div className="px-4 pb-4 space-y-4">
                    {/* Credit Cost */}
                    {feature.creditCost && (
                      <div className="bg-green-500/10 border border-green-500/30 rounded-lg p-3">
                        <p className="text-sm text-green-400">
                          <CreditCard className="w-4 h-4 inline mr-2" />
                          Cost: {typeof feature.creditCost === 'object' 
                            ? Object.entries(feature.creditCost).map(([k, v]) => `${k}: ${v}`).join(', ')
                            : feature.creditCost
                          }
                        </p>
                      </div>
                    )}
                    
                    {/* How to Use */}
                    {feature.howToUse && (
                      <div>
                        <h4 className="font-medium text-white mb-2">How to Use</h4>
                        <ol className="space-y-2">
                          {feature.howToUse.map((step, idx) => (
                            <li key={idx} className="text-sm text-slate-300 flex gap-2">
                              <span className="text-purple-400 font-medium">{idx + 1}.</span>
                              {step.replace(/^\d+\.\s*/, '')}
                            </li>
                          ))}
                        </ol>
                      </div>
                    )}
                    
                    {/* Tips */}
                    {feature.tips && (
                      <div>
                        <h4 className="font-medium text-white mb-2">Tips</h4>
                        <ul className="space-y-1">
                          {feature.tips.map((tip, idx) => (
                            <li key={idx} className="text-sm text-slate-400 flex gap-2">
                              <span className="text-yellow-400">💡</span>
                              {tip}
                            </li>
                          ))}
                        </ul>
                      </div>
                    )}
                    
                    {/* Sub-features */}
                    {feature.subFeatures && (
                      <div>
                        <h4 className="font-medium text-white mb-2">Sub-features</h4>
                        <div className="grid md:grid-cols-2 gap-3">
                          {Object.entries(feature.subFeatures).map(([subKey, subFeature]) => (
                            <div key={subKey} className="bg-slate-800/50 rounded-lg p-3">
                              <h5 className="font-medium text-white text-sm">{subFeature.title}</h5>
                              <p className="text-xs text-slate-400 mt-1">{subFeature.description}</p>
                              <p className="text-xs text-green-400 mt-2">{subFeature.creditCost}</p>
                            </div>
                          ))}
                        </div>
                      </div>
                    )}
                    
                    {/* Includes/Features list */}
                    {feature.includes && (
                      <div>
                        <h4 className="font-medium text-white mb-2">What's Included</h4>
                        <ul className="grid grid-cols-2 gap-2">
                          {feature.includes.map((item, idx) => (
                            <li key={idx} className="text-sm text-slate-400 flex gap-2">
                              <span className="text-green-400">✓</span>
                              {item}
                            </li>
                          ))}
                        </ul>
                      </div>
                    )}
                  </div>
                )}
              </div>
            );
          })}
        </div>

        {/* Account Section */}
        <div className="mt-8 space-y-4">
          <h2 className="text-xl font-bold text-white mb-4">Account & Billing</h2>
          
          {/* Credits */}
          <div className="bg-slate-900/50 border border-slate-800 rounded-xl p-6">
            <h3 className="font-semibold text-white mb-3 flex items-center gap-2">
              <CreditCard className="w-5 h-5 text-purple-400" />
              Understanding Credits
            </h3>
            <div className="space-y-2">
              {manual?.account?.credits?.howItWorks?.map((item, idx) => (
                <p key={idx} className="text-sm text-slate-400">• {item}</p>
              ))}
            </div>
          </div>
          
          {/* Subscription */}
          <div className="bg-slate-900/50 border border-slate-800 rounded-xl p-6">
            <h3 className="font-semibold text-white mb-3 flex items-center gap-2">
              <Shield className="w-5 h-5 text-green-400" />
              Subscription Plans
            </h3>
            <div className="grid md:grid-cols-3 gap-4">
              {manual?.account?.subscription?.plans?.map((plan, idx) => (
                <div key={idx} className="bg-slate-800/50 rounded-lg p-4">
                  <h4 className="font-medium text-white">{plan.name}</h4>
                  <p className="text-purple-400 font-bold">{plan.price}</p>
                  <p className="text-sm text-slate-400">{plan.credits} credits</p>
                </div>
              ))}
            </div>
          </div>
        </div>

        {/* Troubleshooting */}
        <div className="mt-8">
          <h2 className="text-xl font-bold text-white mb-4">Troubleshooting</h2>
          <div className="space-y-3">
            {manual?.troubleshooting?.common_issues?.map((item, idx) => (
              <div key={idx} className="bg-slate-900/50 border border-slate-800 rounded-xl p-4">
                <h4 className="font-medium text-red-400 mb-2">{item.issue}</h4>
                <p className="text-sm text-slate-300">{item.solution}</p>
              </div>
            ))}
          </div>
          
          <div className="mt-4 bg-purple-500/10 border border-purple-500/30 rounded-xl p-4">
            <p className="text-sm text-slate-300">
              Need more help? Contact us at{' '}
              <a href={`mailto:${manual?.troubleshooting?.contact?.email}`} className="text-purple-400 hover:underline">
                {manual?.troubleshooting?.contact?.email}
              </a>
            </p>
          </div>
        </div>
      </main>
    </div>
  );
}
